Smash_Combos.Domain/Models/Infraction.cs

```diff
@@ -0,0 +1,30 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
+using System.Text;
+
+namespace Smash_Combos.Domain.Models
+{
+    public class Infraction
+    {
+        public Infraction()
+        {
+            this.DateInfracted = DateTime.Now;
+        }
+
+        public int Id { get; set; }
+
+        public int UserId { get; set; }
+
+        public int ModeratorId { get; set; }
+
+        public int BanDuration { get; set; }
+
+        public int Points { get; private set; }
```

Comment:
Should this also be nullable?

Reply:
I think when a user gets infracted, they will always get a point, including bans. What do you think?

Reply:
If we have points in bans too then we should keep it that way. But do we need them? Wouldn't a mod see the user's accumulated points and then decide to ban them? Then I don't think we would need to add additional points on the bans themselves.
I'm leaning towards nullable, personally. Good thinking.

Reply:
What if the user gets banned, temporarily. Say 2 weeks. Would they come back with no points for whatever the offense is? or would it make more sense for the moderator to infract AND THEN ban the user?

Reply:
The infraction would still exist, so they would still have the points to their name, even after the ban. The way I thought about it was that the moderator does their work putting out infractions, until they notice that a user has a high amount of infractions/points, and then they issue a ban, setting the length to whatever they feel is long enough. When a user comes back after a ban and gets reported AGAIN, the moderator would see their high point count and would likely ban them again, longer/permanent this time. The bans themselves would not have points, they're just there to _punish_ a user that racks up a lot of points. Was that what you were envisioning?

Reply:
My initial thought was that bans will also give points as a warning would. But honestly, your approach feels much better. I was just trying to reduce the number of steps for UI purposes, but again, this approach feels better. Lets make it nullable.

Reply:
Alright, I'll change it real quick.